src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -171,6 +182,79 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * monitor for new pending txs, add them to a set, so we can monitor when they're confirmed
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
+    // don't fire with nothing connected
+    if (isEmpty(accountSpecifiers)) return
+    // grab the most recent txId
+    const txId = head(txIds)!
+    // grab the actual tx
+    const tx = txsById[txId]
+    // always wear protection, or don't it's your choice really
+    if (!tx) return
+
+    if (tx.caip2 === cosmosChainId) {
+      // cosmos txs only come in when they're confirmed, so refetch that account immediately
+      return refetchAccountByTxId(txId)
+    } else {
+      /**
+       * the unchained getAccount call does not include pending txs in the portfolio
+       * add them to a set, and the two effects below monitor the set on pending txs
+       */
+      if (tx.status === 'pending') setPendingTxIds(new Set([...pendingTxIds, txId]))
+    }
+
+    // txsById changes on each tx - as txs have more confirmations
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch, txIds])
+
+  /**
+   * refetch an account given a newly confirmed txid
+   */
+  const refetchAccountByTxId = useCallback(
+    (txId: TxId) => {
+      // the accountSpecifier the tx came from
+      const { txAccountSpecifier } = deserializeUniqueTxId(txId)
+      // only refetch the specific account for this tx
+      const accountSpecifierMap = accountSpecifiersList.reduce((acc, cur) => {
+        const [chainId, accountSpecifier] = Object.entries(cur)[0]
+        const accountId = chainId + ':' + accountSpecifier
+        if (accountId === txAccountSpecifier) acc[chainId] = accountSpecifier
+        return acc
+      }, {})
+      const { getAccount } = portfolioApi.endpoints
+      dispatch(getAccount.initiate({ accountSpecifierMap }, { forceRefetch: true }))
+    },
+    [accountSpecifiersList, dispatch]
+  )
+
+  /**
+   * monitor the pending tx ids for when they change to confirmed.
+   * when they do change to confirmed, refetch the portfolio for that chain
+   * (unchained does not include assets for pending txs)
+   */
+  useEffect(() => {
+    // don't monitor any of this stuff if we're still loading - txsByIds will be thrashing
+    if (txHistoryStatus !== 'loaded') return
+    if (!pendingTxIds.size) return
+    // can't map a set, spread it first
+    const confirmedTxIds = [...pendingTxIds].filter(txId => txsById[txId]?.status === 'confirmed')
+    // txsById will change often, but we only care that if they've gone from pending -> confirmed
+    if (!confirmedTxIds.length) return
+    // refetch the account for each newly confirmed tx
+    confirmedTxIds.forEach(txId => refetchAccountByTxId(txId))
```

Comment:
Just a thought - It's probably better to rewrite `refetchAccountByTxId` to be `refetchAccount` and use an `accountSpecifier` instead of a `txId` themselves - so we don't end up getting an account multiple times in a row (in case of an account has multiple transactions going from pending to confirmed at the same time).

Reply:
good point, and i did actually consider this as i was writing bit, but i don't think it's actually a major concern in reality